src/rgw/rgw_sal_motr.cc

```diff
@@ -2192,29 +2195,63 @@ int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t en
 
 int MotrObject::get_bucket_dir_ent(const DoutPrefixProvider *dpp, rgw_bucket_dir_entry& ent)
 {
-  int rc = 0;
+  int rc = 0, ret_code = -ENOENT;
```

Comment:
why defaulting ret_code to -ENOENT? keep it 0 only
Also i see ret_code being used for motr calls, so we can rename it to motr_rc